test/ng/httpSpec.js

```diff
@@ -1380,8 +1380,28 @@ describe('$http', function() {
             expect(errCallback.calls.mostRecent().args[0]).toEqualMinErr('$http', 'baddata');
           });
 
-        });
+          it('should not throw an error if JSON is invalid but content-type is not application/json', function() {
+            var errCallback = jasmine.createSpy('error');
+            $httpBackend.expect('GET', '/url').respond('{abcd}', {'Content-Type': 'text/plain'});
+
+            $http.get('/url').then(callback).catch(errCallback);
+            $httpBackend.flush();
+
+            expect(callback).toHaveBeenCalledOnce();
+            expect(errCallback).not.toHaveBeenCalled();
+          });
 
+          it('should return response unprocessed if JSON is invalid but content-type is not application/json', function() {
+            var response = '{abcd}';
+            $httpBackend.expect('GET', '/url').respond(response, {'Content-Type': 'text/plain'});
+
+            $http.get('/url').then(callback);
+            $httpBackend.flush();
+
+            expect(callback.calls.mostRecent().args[0].data).toMatch(response);
```

Comment:
[`toMatch()`][1] is for regular expressions (you can pass a string, but it will be converted to a regular expression). It works in this case, but is unnecessary.

[`toBe()`][2] tests for strict equality (`===`), which works when comparing references to the same object instance or primitives (e.g. strings - as is the case here).

[`toEqual()`][3] can be used for testing loose equality (or with a custom comparator), but it is not necessary here.

[1]: https://github.com/jasmine/jasmine/blob/9cb2f06aa665870dbd5df5fd58a35863738961b3/src/core/matchers/toMatch.js#L14-L28
[2]: https://github.com/jasmine/jasmine/blob/9cb2f06aa665870dbd5df5fd58a35863738961b3/src/core/matchers/toMatch.js#L14-L28
[3]: https://github.com/jasmine/jasmine/blob/9cb2f06aa665870dbd5df5fd58a35863738961b3/src/core/matchers/toEqual.js#L10-L28